### PR TITLE
doc: scripts: gen_kconfig_rest: fix rendering issues

### PR DESCRIPTION
**Pull request by gmarull**
As of today we assume that the help text contained in the Kconfig help
field will be properly formatted RST. However, there is no guarantee
that this will happen. This patch tries to address this problem by:

- Escaping inline RST characters for text that needs to be rendered as
  "normal text" (i.e. prompt)
- Use code-blocks to render help content. This way help is rendered as
  literal text and so it does not get processed by the RST parser.

A secondary change: the index table now contains the prompt information
only. The page was taking a long time to render because of the large
amount of content. Using a table doesn't help either. Note that adding
code-block to include help as a literal increases build time even more,
so it is not an option.

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
Can you share some screenshots of the before and after showing the fix?

**Comment by gmarull** [comment]
One example:

before:
image

after:
image

**Review by carlescufi — APPROVED** [review]
LGTM

File: doc/_scripts/gen_kconfig_rest.py [hunk 1]
```diff
@@ -35,6 +36,10 @@
 
 """
 
+def escape_inline_rst(text):
+    # Escape reStructuredText inline markup characters
+    return re.sub(r"(\*|_|`)", r"\\\1", text)
```
Comment (mbolivar-nordic) [inline doc/_scripts/gen_kconfig_rest.py:41]: Should this be r'\\1'? Why the extra backslash?

Comment (mbolivar-nordic) [reply doc/_scripts/gen_kconfig_rest.py:41]: ah, OK, I see now. re.sub() processes backslash escapes in the replacement.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**